lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -327,87 +318,101 @@ void SpecialFunctionHandler::handleAssert(ExecutionState &state,
       Executor::Assert);
 }
 
-void SpecialFunctionHandler::handleAssertFail(ExecutionState &state,
-                                              KInstruction *target,
-                                              std::vector<ref<Expr> > &arguments) {
-  assert(arguments.size()==4 && "invalid number of arguments to __assert_fail");
-  executor.terminateStateOnError(
-      state, "ASSERTION FAIL: " + readStringAtAddress(state, arguments[0]),
-      Executor::Assert);
+void
+SpecialFunctionHandler::handleAssertFail(ExecutionState &state,
+                                         KInstruction *target,
+                                         std::vector<ref<Expr> > &arguments) {
+  assert(arguments.size() == 4 &&
+         "invalid number of arguments to __assert_fail");
+  if (INTERPOLATION_ENABLED && SpecTypeToUse != NO_SPEC &&
+      state.txTreeNode->isSpeculationNode()) {
+    executor.terminateStateOnError(
```

Comment:
This should be printed if debug-subsumption >= 1.

Reply:
this is not print, it is to call executor.terminateStateOnError when error, is that right?

Reply:
can you please check if the speculation failure message can be printed in debug-subsumption >= 1 in function terminateStateOnError

Reply:
yes, it is printed.
the same as we print ASSERTION FAIL but this failure is in speculation tree.